miso-web/src/main/webapp/scripts/library_hot.js

```diff
@@ -2,834 +2,402 @@
+    return {
+      header : 'Index ' + n,
+      data : 'index' + n + 'Label',
+      type : 'autocomplete',
+      strict : true,
+      filter : false,
+      allowInvalid : true,
+      trimDropdown : false,
+      source : [ '' ],
+      include : true,
+      unpack : function(lib, flat, setCellMeta) {
+        var id = lib['index' + n + 'Id'];
+        flat['index' + n + 'Label'] = id ? Constants.indexFamilies.reduce(
+            function(acc, family) {
+              return family.indices.reduce(function(acc, index) {
+                return id == index.id ? index.label : acc;
+              }, acc);
+            }, null) : 'No index';
+      },
+      pack : function(lib, flat, errorHandler) {
+        var label = flat['index' + n + 'Label'];
+        var families = Constants.indexFamilies.filter(function(family) {
+          return family.name == flat.indexFamilyName;
+        });
+        lib['index' + n + 'Id'] = families.length == 0 ? null : Utils.array
+            .maybeGetProperty(Utils.array.findFirstOrNull(function(index) {
+              return index.label == label;
+            }, families[0].indices), 'id');
+      },
+      depends : 'indexFamilyName',
+      update : function(lib, flat, value, setReadOnly, setOptions, setData) {
+        var pt = getPlatformType(flat.platformType);
+        var indices = (Utils.array.maybeGetProperty(Utils.array
+            .findFirstOrNull(function(family) {
+              return family.name == value && family.platformType == pt;
+            }, Constants.indexFamilies), 'indices') || []).filter(
+            function(index) {
+              return index.position == n;
+            }).map(function(index) {
+          return index.label;
+        }).sort();
+        var data;
+        if (indices.length == 0 || n > 1) {
+          indices.unshift('No index');
+          data = 'No index';
+        } else {
+          data = '';
         }
+        setOptions(indices);
+        setData(data);
       }
-      return lib;
-    });
-  },
-
-  /**
-   * Modifies attributes of LibraryDtos so Handsontable displays them for edit
-   */
-  prepLibrariesForEdit: function (libraries) {
-    return libraries.map(function (lib) {
-      lib.librarySelectionTypeAlias = Hot.getAliasFromId(lib.librarySelectionTypeId, Hot.dropdownRef.selectionTypes) || '(None)';
-      lib.libraryStrategyTypeAlias = Hot.getAliasFromId(lib.libraryStrategyTypeId, Hot.dropdownRef.strategyTypes) || '(None)';
-      if (Hot.detailedSample) {
-        // if any members are null, fill them with empty objects otherwise things go poorly
+    };
+  };
+  
+  var makeDesignUpdate = function(id, nameProperty, defaultValue, source) {
+    return function(lib, flat, value, setReadOnly, setOptions, setCell) {
+      var design = Utils.array.findFirstOrNull(
+          Utils.array.namePredicate(value), Constants.libraryDesigns);
+      if (design) {
+        setCell(Utils.array.maybeGetProperty(Utils.array.findFirstOrNull(
```

Comment:
It sets the cell contents: https://github.com/TGAC/miso-lims/pull/881/files#diff-06f0aa9b486ef785656b9d1f0d7afd6fR272